plugins/swiotrefactor/src/ad74413r/currentchnlinfo.cpp

```diff
@@ -0,0 +1,24 @@
+#include "ad74413r/currentchnlinfo.h"
+
+using namespace scopy::swiotrefactor;
+CurrentChnlInfo::CurrentChnlInfo(QString plotUm, QString hwUm, iio_channel *iioChnl, CommandQueue *cmdQueue)
+	: ChnlInfo(plotUm, hwUm, iioChnl, cmdQueue)
+{
+	m_rangeValues = {MIN_CURRENT_VALUE, MAX_CURRENT_VALUE};
+}
+
+CurrentChnlInfo::~CurrentChnlInfo() {}
+
+double CurrentChnlInfo::convertData(unsigned int data)
+{
+	double convertedData = 0.0;
+	double defaultFactor = m_unitOfMeasureFactor.contains(m_hwUm.at(0)) ? m_unitOfMeasureFactor[m_hwUm.at(0)] : 1;
+	double newFactor = m_unitOfMeasureFactor.contains(m_plotUm.at(0)) ? m_unitOfMeasureFactor[m_plotUm.at(0)] : 1;
+	double factor = defaultFactor / newFactor;
```

Comment:
digital_input and digital_input_loop channels need to be treated differently and shouldn't have a scale factor based on units (this causes wrong data interpretation due to 1(high) becoming 0.1 mV